static/app/components/compactSelect/control.tsx

```diff
@@ -246,6 +247,12 @@ export function Control({
         // Wait for overlay to appear/disappear
         await new Promise(resolve => resolve(null));
 
+        // Focus on search box if present
+        if (searchable) {
+          searchRef.current?.focus();
+          return;
+        }
+
```

Comment:
If present (`searchable` is true), the search box will receive focus when the menu opens. Otherwise, the list box will receive focus.

![image](https://user-images.githubusercontent.com/44172267/228975138-fc0e4667-4a80-44e4-972a-ebb9a682cbcf.png)